src/User.js

```diff
@@ -7,46 +8,72 @@ class User {
   constructor(name, id, pantry) {
     this.name = name;
     this.id = id;
-    this.pantry = pantry;
+    this.pantry = new Pantry(pantry);
     this.favoriteRecipes = [];
     this.recipesToCook = [];
   }
 
-  addFavoriteRecipe(id) {
-    if (!this.favoriteRecipes.includes(id)) {
-      this.favoriteRecipes.push(id);
+  addFavoriteRecipe(recipeToAdd) {
+    const recipeExists = this.findRecipe(this.favoriteRecipes, recipeToAdd);
+    if (!recipeExists) {
+      this.favoriteRecipes.push(recipeToAdd);
     }
-    // issue # 23
   }
 
-  removeFavoriteRecipe(id) {
-    let item = this.favoriteRecipes.indexOf(id);
-    this.favoriteRecipes.splice(item, 1);
+  addRecipesToCook(recipeToAdd) {
+    const recipeExists = this.findRecipe(this.recipesToCook, recipeToAdd);
+    if (!recipeExists) {
+      this.recipesToCook.push(recipeToAdd);
+    }
+  }
+
+  removeFavoriteRecipe(recipeToRemove) {
+    // let item = this.favoriteRecipes.indexOf(recipe.id);
+    const targetIndex = this.findRecipeIndex(this.favoriteRecipes, recipeToRemove);    
+    this.favoriteRecipes.splice(targetIndex, 1);
     // #19
   }
 
-  addRecipesToCook(id) {
-    if (!this.recipesToCook.includes(id)) {
-      this.recipesToCook.push(id);
+  addRecipesToCook(recipeToAdd) {
+    const recipeExists = this.findRecipe(this.recipesToCook, recipeToAdd);
+    if (!recipeExists) {
+      this.recipesToCook.push(recipeToAdd);
     }
-    //#20
   }
 
-  removeRecipesToCook(id) {
-    let item = this.recipesToCook.indexOf(id);
-    this.recipesToCook.splice(item, 1);
+  findRecipe(recipes, recipeToFind) {
+    return recipes.find(recipe => {
+      return recipe.id === recipeToFind.id;
+    });
+  }
+
+  findRecipeIndex(recipes, recipeToFind) {
+    return recipes.findIndex(recipe => {
+      return recipe.id === recipeToFind.id;
+    });
   }
-  // #21
 
+  removeRecipesToCook(recipeToRemove) {
+    const targetIndex = this.findRecipeIndex(this.recipesToCook, recipeToRemove);
+    this.recipesToCook.splice(targetIndex, 1);
+  }
+  // #21
+  getSavedRecipes() {
+    // how to get unique values from an array
+    const savedRecipesWithDuplicates = this.favoriteRecipes.concat(this.recipesToCook);
+    const uniqueRecipes = new Set(savedRecipesWithDuplicates);
+    //need to get array from set
+    return Array.from(uniqueRecipes);
+  }
   //search recipesToCook OR favoriteRecipes by type
-  findRecipesByType(savedRecipes, tagName) {
-    return savedRecipes.filter((recipe) => {
+  findRecipesByType(tagName) {
```

Comment:
Could we make all of the findRecipesBy... methods work dynamically taking in any list of recipes? I am thinking about the ability for them to search in the entire recipes dataset.